src/client.ts

```diff
@@ -347,26 +349,30 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
     return await this.userContext.fetchQualifiedSegments(options);
   }
 
-  public setUser(userInfo: UserInfo): void {
-    // TODO add check for valid user
-    if (userInfo.id) {
-      this.user.id = userInfo.id;
-      this.isUserReady = true;
+  public async setUser(userInfo: UserInfo): Promise<void> {
+    this.isUserPromiseResolved = false;
+    this.isUserReady = true;
 
-      if (this._client) {
-        this.userContext = this._client.createUserContext(userInfo.id, userInfo.attributes);
-      } else {
-        logger.warn(
-          'Unable to create user context for user id "%s" because Optimizely client failed to initialize.',
-          this.user.id
-        );
-      }
+    //reset user info
+    this.user = default_user;
+
+    this.user.id = userInfo.id;
+
+    if (this._client) {
+      this.userContext = this._client.createUserContext(userInfo.id ?? undefined, userInfo.attributes);
+    } else {
+      logger.warn(
+        'Unable to create user context for user id "%s" because Optimizely client failed to initialize.',
+        this.user.id
+      );
     }
 
     if (userInfo.attributes) {
       this.user.attributes = userInfo.attributes;
     }
 
+    await this.fetchQualifiedSegments();
```

Comment:
Error handling? 
Log error message if this returns false? Should we continue on error?

Reply:
We also need support for options for fetchQualifedSegments.

Reply:
the error handling for fetchqualifiedsegment is done independently in the fetchqualifiedsegment method itself. 

for the options support, since setuser may be a new user everytime, options dont have any purpose here. If its a new user a new calll is made instead of reading from the cache, the userid being the deciding key here.

Reply:
I understand all errors logged in js core. Wondering we should continue to resolve the promise when we get false for `fetchQualifedSegments`?